tests/nrf5340_audio/lc3_file/src/main.c

```diff
@@ -0,0 +1,348 @@
+
+	ret = lc3_file_open(&file, "test.lc3");
+	zassert_equal(0, ret, "lc3_file_open() should return 0");
+
+	ret = lc3_file_frame_get(&file, frame_buffer, sizeof(frame_buffer));
+	zassert_equal(-EIO, ret, "lc3_file_frame_get() should return -EIO");
+}
+
+ZTEST(lc3_file, test_lc3_file_open)
+{
+	int ret;
+	struct lc3_file_ctx file;
+
+	sd_card_read_fake.custom_fake = sd_card_read_lc3_file_fake;
+
+	ret = lc3_file_open(&file, "test.lc3");
+
+	zassert_equal(0, ret, "lc3_file_open() should return 0");
+	zassert_equal(1, sd_card_read_fake.call_count, "sd_card_read() should be called once");
+
+	zassert_equal(0xcc1c, file.lc3_header.file_id, "File ID is wrong (is 0x%2x)",
+		      file.lc3_header.file_id);
+	zassert_equal(0x0012, file.lc3_header.hdr_size, "Header size is wrong, is 0x%2x",
+		      file.lc3_header.hdr_size);
+	zassert_equal(0x01e0, file.lc3_header.sample_rate, "Sample rate is wrong, is 0x%2x",
+		      file.lc3_header.sample_rate);
+	zassert_equal(0x0140, file.lc3_header.bit_rate, "Bit rate is wrong, is 0x%2x",
+		      file.lc3_header.bit_rate);
+	zassert_equal(0x0001, file.lc3_header.channels, "Number of channels is wrong, is 0x%2x",
+		      file.lc3_header.channels);
+	zassert_equal(0x03e8, file.lc3_header.frame_duration, "Frame duration is wrong, is 0x%2x",
+		      file.lc3_header.frame_duration);
+	zassert_equal(0x0000, file.lc3_header.rfu, "RFU is wrong, is 0x%2x", file.lc3_header.rfu);
+	zassert_equal(0x0780, file.lc3_header.signal_len_lsb,
+		      "Signal length LSB is wrong, is 0x%2x", file.lc3_header.signal_len_lsb);
+	zassert_equal(0x0000, file.lc3_header.signal_len_msb,
+		      "Signal length MSB is wrong, is 0x%2x", file.lc3_header.signal_len_msb);
+}
+
+ZTEST(lc3_file, test_lc3_file_open_invalid_header)
+{
+	int ret;
+	struct lc3_file_ctx file;
+
+	sd_card_read_fake.custom_fake = sd_card_read_header_fake_invalid;
+
+	ret = lc3_file_open(&file, "test.lc3");
+
+	zassert_equal(-EINVAL, ret, "lc3_file_open() should return -EINVAL");
+	zassert_equal(1, sd_card_read_fake.call_count, "sd_card_read() should be called once");
+}
+
+ZTEST(lc3_file, test_lc3_file_open_invalid_sd_card_open_failure)
+{
+	int ret;
+	struct lc3_file_ctx file;
+
+	sd_card_open_fake.return_val = -EINVAL;
+
+	ret = lc3_file_open(&file, "test.lc3");
+
+	zassert_equal(-EINVAL, ret, "lc3_file_open() should return -EINVAL");
+	zassert_equal(1, sd_card_open_fake.call_count, "sd_card_open() should be called once");
+}
+
+ZTEST(lc3_file, test_lc3_file_open_invalid_sd_card_read_failure)
+{
+	int ret;
+	struct lc3_file_ctx file;
+
+	sd_card_read_fake.return_val = -EINVAL;
+
+	ret = lc3_file_open(&file, "test.lc3");
+
+	zassert_equal(-EINVAL, ret, "lc3_file_open() should return -EINVAL");
+	zassert_equal(1, sd_card_read_fake.call_count, "sd_card_read() should be called once");
+}
+
+ZTEST(lc3_file, test_lc3_file_close)
+{
+	int ret;
+	struct lc3_file_ctx file;
+
+	ret = lc3_file_close(&file);
+
+	zassert_equal(0, ret, "lc3_file_close() should return success");
```

Comment:
Shouldn't this return `-EPERM` since there is no open before this?

Reply:
The underlying API probably does, but this functionality gets a bit "lost" in the mocking. lc3_file_close does nothing to check whether a file has been opened, but as long as the underlying API does the error will be forwarded, as is tested by `test_lc3_file_close_invalid`.

It could be a discussion if the `lc3_file` module should check this though. Right now I can't think of this being something that must be guarded against as only the calls to `fs` (probably) must be initialized correctly before using the objects, which should be checked in that API.